bulk-model-sync-gradle/src/main/kotlin/org/modelix/model/sync/bulk/gradle/ModelSyncGradlePlugin.kt

```diff
@@ -0,0 +1,234 @@
+/*
+ * Copyright (c) 2023.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.modelix.model.sync.bulk.gradle
+
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.api.tasks.Sync
+import org.gradle.api.tasks.TaskProvider
+import org.modelix.model.sync.bulk.gradle.config.LocalSource
+import org.modelix.model.sync.bulk.gradle.config.LocalTarget
+import org.modelix.model.sync.bulk.gradle.config.ModelSyncGradleSettings
+import org.modelix.model.sync.bulk.gradle.config.ServerSource
+import org.modelix.model.sync.bulk.gradle.config.ServerTarget
+import org.modelix.model.sync.bulk.gradle.config.SyncDirection
+import org.modelix.model.sync.bulk.gradle.tasks.ExportFromModelServer
+import org.modelix.model.sync.bulk.gradle.tasks.ExportFromMps
+import org.modelix.model.sync.bulk.gradle.tasks.GenerateAntScriptForMps
+import org.modelix.model.sync.bulk.gradle.tasks.ImportIntoModelServer
+import org.modelix.model.sync.bulk.gradle.tasks.ImportIntoMps
+import org.modelix.model.sync.bulk.gradle.tasks.ValidateSyncSettings
+import java.io.File
+import java.util.Properties
+
+class ModelSyncGradlePlugin : Plugin<Project> {
+
+    private lateinit var settings: ModelSyncGradleSettings
+
+    override fun apply(project: Project) {
+        settings = project.extensions.create("modelSync", ModelSyncGradleSettings::class.java)
+        getBaseDir(project).mkdirs()
+
+        project.afterEvaluate {
+            val validateSyncSettings = project.tasks.register("validateSyncSettings", ValidateSyncSettings::class.java) {
+                settings.taskDependencies.forEach { dependency ->
+                    it.dependsOn(dependency)
+                }
+                it.settings.set(settings)
+            }
+            val modelixCoreVersion = readModelixCoreVersion() ?: throw RuntimeException("modelix.core version not found")
```

Comment:
We could probably give the user of the plugin a better hint if there's anything he can do to avoid this exception of if this should have never happened.